homeassistant/components/spc/__init__.py

```diff
@@ -16,61 +14,60 @@
 CONF_API_URL = "api_url"
 
 DOMAIN = "spc"
-DATA_API = "spc_api"
+DATA_KEY = "spc"
 
 SIGNAL_UPDATE_ALARM = "spc_update_alarm_{}"
 SIGNAL_UPDATE_SENSOR = "spc_update_sensor_{}"
-
-CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.Schema(
-            {
-                vol.Required(CONF_WS_URL): cv.string,
-                vol.Required(CONF_API_URL): cv.string,
-            }
-        )
-    },
-    extra=vol.ALLOW_EXTRA,
-)
+PLATFORMS = ["alarm_control_panel", "binary_sensor"]
 
 
 async def async_setup(hass, config):
+    """Old way to set up SPC devices."""
+    return True
+
+
+async def async_setup_entry(hass, entry):
     """Set up the SPC component."""
 
-    async def async_upate_callback(spc_object):
+    await async_setup_internal(hass, entry)
```

Comment:
Why is this a separate coroutine if it's only called once?

Reply:
Thanks, valid point. Fixed!